apps/extension/src/router/types/router.ts

```diff
@@ -0,0 +1,44 @@
+import { Message } from "../Message";
+
+export type MessageSender = Pick<
+  browser.runtime.MessageSender,
+  "id" | "url" | "tab"
+>;
+
+export interface Env {
+  readonly isInternalMsg: boolean;
+  readonly requestInteraction: any;
+}
+
+export type FnRequestInteraction = <M extends Message<unknown>>(
```

Comment:
Is this like a dynamic `Message`? It seems as if it is not used outside of this file for now.

Reply:
It's not currently used, but it will be implemented when we are requesting to connect to the extension, and this will need to be in place.